hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -148,12 +148,10 @@ public void run() {
           }
         }
       } catch (IOException e) { // stream related
+        LOG.error("Failed to read stream of replication entries", e);
```

Comment:
We already log about the EOF issue inside _handleEofException_, so maybe worth keep this one as debug to avoid flooding the log while we keep retrying?

Reply:
+1

Reply:
@wchevreuil  We will log inside handleEofException *only* if the exception or cause of exception is an EOF not otherwise. If we make this line at debug then we won't see any logs if it is caused by non-eof exception.

Reply:
Thanks for reviewing.
In our use, we got a case, the read position is greater than the WAL length, reader thread stuck in retry and no log. We haven't figure out the case, maybe it's a bug. So hope to print out the error log in time first when unexpected IOE happens.
So how about logging only when IOE can’t be handled by `handleEofException`? Only log in unexpected cases?
I just fixed it, please take another review.

Reply:
That's even better! Thanks for thinking on this further, @ddupg !